### PR TITLE
Replace netstandard1.0 and netstandard1.3 with netstandard2.0

### PR DESCRIPTION
**Pull request by Evangelink**
## Description

Target .netstandard2.0 instead of netstandard1.0/netstandard1.3

## Related issue

Fixes [AB#1585589](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1585589)

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -1273,10 +1219,8 @@ function Build-SpecificProjects {
     # FrameworksAndOutDirs format ("<target_framework>", "<output_dir>").
     $FrameworksAndOutDirs = (
         ("net462", "net462\win7-x64"),
-        ("netstandard1.0", "netstandard1.0"),
-        ("netstandard1.3", "netstandard1.3"),
         # REVIEW ME: Why do we copy netstandard2.0 into netcorecorapp2.1?
-        ("netstandard2.0", "netcoreapp2.1"),
+        ("netstandard2.0", "netstandard2.0"),
```
Comment (Evangelink) [inline scripts/build.ps1:1223]: I have fixed this copy folder. If there was a good reason to copy netstandard dlls into netcoreapp2.1 folder, please let me know. I will revert and add a comment.

File: src/Microsoft.TestPlatform.AdapterUtilities/PublicAPI/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -15,6 +15,7 @@ Microsoft.TestPlatform.AdapterUtilities.HierarchyConstants.Levels
 Microsoft.TestPlatform.AdapterUtilities.ManagedNameConstants
 Microsoft.TestPlatform.AdapterUtilities.ManagedNameUtilities.InvalidManagedNameException
 Microsoft.TestPlatform.AdapterUtilities.ManagedNameUtilities.InvalidManagedNameException.InvalidManagedNameException(string? message) -> void
+Microsoft.TestPlatform.AdapterUtilities.ManagedNameUtilities.InvalidManagedNameException.InvalidManagedNameException(System.Runtime.Serialization.SerializationInfo! info, System.Runtime.Serialization.StreamingContext context) -> void
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.AdapterUtilities/PublicAPI/PublicAPI.Shipped.txt:18]: Moved from net/PublicAPI.Shipped.txt

File: src/Microsoft.TestPlatform.CoreUtilities/Tracing/EqtTrace.cs [hunk 3]
```diff
@@ -822,12 +822,6 @@ private static void WriteAtLevel(PlatformTraceLevel level, string format, params
 
     private static void FailDebugger(string? message)
     {
-#if DEBUG
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.CoreUtilities/Tracing/EqtTrace.cs:825]: No need to keep this debug pragma as it's already included in `Debug.Fail`.

File: src/Microsoft.TestPlatform.ObjectModel/TraitCollection.cs [hunk 4]
```diff
@@ -11,28 +11,23 @@ namespace Microsoft.VisualStudio.TestPlatform.ObjectModel;
 /// <summary>
 /// Class that holds collection of traits
 /// </summary>
-#if NETFRAMEWORK
+#if NETFRAMEWORK // REVIEW ME: This could be enabled for netcore and netstandard
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.ObjectModel/TraitCollection.cs:14]: Not for this PR but is there any reason not to enable this for all tfms? The ones we have all support it.

File: src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec [hunk 5]
```diff
@@ -19,14 +19,6 @@
                 branch="$BranchName$"
                 commit="$CommitId$" />
     <dependencies>
-      <group targetFramework="uap10.0">
```
Comment (Evangelink) [inline src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec:22]: Shall we add `netstandard2.0` in replacement of `uap10.0`?

File: src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec [hunk 6]
```diff
@@ -24,13 +24,12 @@
         <dependency id="Newtonsoft.Json" version="$JsonNetVersion$"/>
         <dependency id="Microsoft.TestPlatform.ObjectModel" version="$Version$"/>
       </group>
-      <group targetFramework="netstandard1.5">
-        <dependency id="System.Collections.NonGeneric" version="[4.0.1, )" />
-        <dependency id="System.Security.Principal.Windows" version="[4.0.1, )" />
-        <dependency id="System.Collections.Specialized" version="[4.0.1, )" />
+      <group targetFramework="netstandard2.0">
+        <!-- REVIEW ME: Why don't we have dependency to Newtonsoft.Json here?  -->
```
Comment (Evangelink) [inline src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec:28]: Other TFMs are declaring dependency on `Newtonsoft.Json`. Why are we not doing it for `netstandard2.0`?

File: src/package/nuspec/TestPlatform.TestHost.nuspec [hunk 7]
```diff
@@ -25,12 +25,6 @@
         <dependency id="Newtonsoft.Json" version="$JsonNetVersion$"/>
       </group>
 
-      <group targetFramework="uap10.0">
```
Comment (Evangelink) [inline src/package/nuspec/TestPlatform.TestHost.nuspec:28]: I don't think it makes sense to have netstandard2.0 for testhost but then I am not sure what to ship for uap?

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs [hunk 8]
```diff
@@ -600,7 +598,7 @@ public static RunConfiguration FromXml(XmlReader reader)
                         XmlRunSettingsUtilities.ThrowOnHasAttributes(reader);
                         string collectSourceInformationStr = reader.ReadElementContentAsString();
 
-                        bool bCollectSourceInformation = true;
+                        bool bCollectSourceInformation;
```
Comment (MarcoRossignoli) [inline src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs:601]: if parse fail should this be `true`?

File: src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec (hunk 5 above)
Comment (MarcoRossignoli) [reply src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec:22]: don't we have netstandard2.0 in the layout?

**Comment by MarcoRossignoli** [comment]
@nohwnd there's UWP discussion open I think, what to ship in the nuget.

File: src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec (hunk 6 above)
Comment (nohwnd) [reply src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec:28]: That looks like an error.

File: src/package/nuspec/TestPlatform.TestHost.nuspec (hunk 7 above)
Comment (nohwnd) [reply src/package/nuspec/TestPlatform.TestHost.nuspec:28]: Why it does not make sense to have netstandard2.0 for testhost?

**Review by nohwnd — APPROVED** [review]

**Review by nohwnd — CHANGES_REQUESTED** [review]
Withdrawing my approval, so we can finish the uwp discussion.

File: src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs (hunk 8 above)
Comment (Evangelink) [reply src/Microsoft.TestPlatform.ObjectModel/RunSettings/RunConfiguration.cs:601]: If parse fails we throw an exception (this is only a local variable not a field).

File: src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec (hunk 5 above)
Comment (Evangelink) [reply src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec:22]: I have just checked and we have only `net462`, `netcoreapp3.1` and before `uap10.0`. That's why I think adding `netstandard2.0` would be useful.

File: src/package/nuspec/TestPlatform.TestHost.nuspec (hunk 7 above)
Comment (Evangelink) [reply src/package/nuspec/TestPlatform.TestHost.nuspec:28]: As we don't target netcoreapp, I thought this package is meant for specific scenario and so `netstandard2.0` seemed too generic.

**Review by Haplois — COMMENTED** [review]
IMO, we should keep `netstandard1.3` too. 

If we change our baseline to `netstandard2.0` we will lose UAP support for Windows 10 before RS3 (1709), the issue here is Windows 10 RS1 LTSC is supported until October 2026, and Windows 10 TH1 LTSC is supported until October 2025.

Said versions are already retired in general access branches, so I am OK either way. 

So vote
 - 👍  for removal of `netstandard1.3` too.
 - 👎  for keeping `netstandard1.3`, and removing `netstandard1.0` only.

**Comment by Evangelink** [comment]
For the record, the problem wouldn't be on TP side as it's not injected onto UWP but problem will happen for Test Frameworks. For example, we won't be able to ship new versions of MSTest still supporting these old versions of UWP as we will no longer have the correct netstandard for ObjectModel and related objects.

@Haplois to correct me if I misunderstood your concern.

**Comment by MarcoRossignoli** [comment]
> Said versions are already retired in general access branches

what does it meam?Can you elaborate?

**Comment by Haplois** [comment]
> > Said versions are already retired in general access branches
> 
> what does it mean?Can you elaborate?

They are already retired in consumer and enterprise editions, only supported channel is LTSC.

File: src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec (hunk 5 above)
Comment (Evangelink) [reply src/package/nuspec/Microsoft.NET.Test.Sdk.nuspec:22]: @nohwnd Any guidance here?

File: src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec (hunk 6 above)
Comment (Evangelink) [reply src/package/nuspec/TestPlatform.Extensions.TrxLogger.nuspec:28]: I will fix this one as a follow-up PR so things don't get mixed up.

**Comment by Evangelink** [comment]
We discussed the issue in a team meeting and decided to move forward with this change.

@Haplois @MarcoRossignoli @nohwnd I'd like to have 3 approvals on this one if you have some time.

**Review by Haplois — APPROVED** [review]

**Review by Haplois — APPROVED** [review]

**Review by nohwnd — APPROVED** [review]

**Pull request merged**